### PR TITLE
Port signature utils to sdk from agent

### PR DESCRIPTION
**Pull request by Itxaka**
It was added first o agent as it was needed for upgrade but if we are gonna reuse this to check for signature validity in random files, we better have it on sdk so we can use it both on agent and immucore and works exactly the same

### CODE REVIEW

File: Earthfile [hunk 1]
```diff
@@ -15,22 +15,23 @@ go-deps:
     ARG GO_VERSION
     FROM golang:$GO_VERSION-alpine
     WORKDIR /build
-    COPY go.mod go.sum ./
+    COPY . .
+    RUN go mod tidy
```
Comment (jimmykarily) [inline Earthfile:19]: I think this was for caching the step of downloading modules, but I'm not sure. I guess the next step will be cached so  this is not needed?

Comment (jimmykarily) [reply Earthfile:19]: Hmm, `COPY . .` cache will always be invalidated, for any changed file. While with the previous code, only changes in `go.{mod,sum}` would invalidate the cache, no?

**Comment by jimmykarily** [comment]
This is just a "copy over" PR right?

**Review by jimmykarily — APPROVED** [review]

File: Earthfile (hunk 1 above)
Comment (Itxaka) [reply Earthfile:19]: Yes indeed but this has been giving us issues, especially with the maintenance of keeping go 1.19 around. So we changed this in all other repos to work around some issues, so I just copy it here so it's the same all around our repos.

**Pull request merged**